src/auth/auth.ts

```diff
@@ -0,0 +1,449 @@
+import {
+  getConfig,
+  GleanConfig,
+  GleanOAuthConfig,
+  GleanTokenConfig,
+  isBasicConfig,
+  isTokenConfig,
+} from '../config/config.js';
+import open from 'open';
+import { debug, error, trace } from '../log/logger.js';
+import readline from 'node:readline';
+import { loadTokens, saveTokens, Tokens } from './token-store.js';
+import {
+  AuthResponse,
+  isAuthResponse,
+  isTokenSuccess,
+  TokenError,
+  TokenResponse,
+} from './types.js';
+import { saveOAuthMetadata } from './oauth-cache.js';
+import { AuthError } from './error.js';
+
+/**
+ * Validate that the configuration can plausibly access the resource.  This
+ * means that either a Glean token was provided, or we have enough information
+ * to attempt an OAuth flow.
+ *
+ * If there's no token, OAuth discover will occur (i.e. requests to oauth
```

Comment:
1/5 (nit, non-blocking)
```suggestion
 * If there's no token, OAuth discovery will occur (i.e. requests to OAuth
```

(unless "OAuth discover" is a specific term of art here)